### ISSUE TITLE
Feature Idea: Link to another card

### ISSUE THREAD

**Issue by chenilim** [mattermost/focalboard#324]
**Summary:**
Add a content type that is a link to another card. The link should have a "link type" e.g.
* (General / default) Is related to
* Depends on

**How important this is to me and why:**

Importance: Medium

Use cases:
1. Link a main task card to a set of subtask cards
2. Link to related cards

**Additional context / similar features:**
Should be able to link to cards on other boards too.

Bonus: Cards that are linked to from other cards show the backlinks.

**Comment by chenilim** [comment]
Also did a quick search, and both [Trello](https://help.trello.com/article/1165-task-dependencies) and [Notion](https://www.notion.so/Create-links-backlinks-cb3c406403c940b68af52de121e05474) support links to other cards / pages, but mainly in the form of a URL with minimal unfurling.

I think we could potentially go a step further by giving links an optional "link type". E.g. this could make building dependency graphs easier in the future.

Has anyone seen other and better ways of handling links between tasks or dependencies?

**Comment by mblennegard** [comment]
Another neat thing to have for these links would be the ability to open the link in a sidebar or similar, like in Roam Research, to being able view both cards side by side.

**Comment by LimitlessDonald** [comment]
Hi guys, thanks for your work on this cool software. 
ANy progress with this please?

**Comment by chenilim** [comment]
Hi @LimitlessDonald, thanks for asking. Yes, short answer is we expect to start working on card links and dependencies towards the middle of this year. In the meantime, if you or anyone else has specific use-cases in mind, please share, as that will help us prioritize features. Thanks! cc @wuwinson.

**Comment by digitalirony** [comment]
@chenilim For my specific use case is:
Master card/ticket with sub tasks/tickets/cards.
The master card might have option to set bool/checkbox on sub tasks dependant on their status/progress.

**Comment by Oobiewan** [comment]
Hi,
I would love to be able to link tickets to each other, specifically to show a "depends on" relationship. This would allow one to see if a specific task can be started or another one must be completed first.

**Comment by wuwinson** [comment]
Thanks for your feedback everyone! Card dependencies is under consideration as part of our [cross-team collaboration](https://www.loom.com/share/873f0aead81d46b4bb97ef859423ac69) features. We don't have an ETA on when this will land, but would appreciate any [additional feedback](https://forms.gle/fMZA8qTZgPuiBr2NA) you might have. Thank you!

**Comment by pauleaster** [comment]
Hi all,
I have just started to use focalboard for personal sprint organisation and one of the first things that I was looking for was a link to different cards. So I will add my voice to this ticket. In the meantime, is there are work around if we clone focalboard ourselves?

**Comment by wuwinson** [comment]
Thanks for the feedback @pauleaster! There are some work arounds that aren't ideal, but may help depending on how many items you want to link:

- [Checkboxes in card descriptions](https://docs.mattermost.com/boards/work-with-cards.html#card-descriptions): If you have multiple subtasks, can list them out as checkbox items within the card description. [Card badges](https://docs.mattermost.com/boards/work-with-cards.html#card-badges) on kanban view will help you quickly identify the status of those checkboxes within each card.
- [URL properties](https://docs.mattermost.com/boards/work-with-cards.html#work-with-property-types) will allow you to link any URL to your card. You can [copy the card URL](https://docs.mattermost.com/boards/share-and-collaborate.html#share-cards-on-channels) and paste it on this property to "link" the cards. Add multiple URL properties to link more items to the card.
- [Commenting on a card](https://docs.mattermost.com/boards/work-with-cards.html#comment-on-a-card) will work similarly to the URL property, but gives you more flexibility on the number of items you can link. Downside is that the links will get mixed in with other comments and cannot be managed like a property.

Hope that helps!

**Comment by marcungeschikts** [comment]
> * [URL properties](https://docs.mattermost.com/boards/work-with-cards.html#work-with-property-types) will allow you to link any URL to your card. You can [copy the card URL](https://docs.mattermost.com/boards/share-and-collaborate.html#share-cards-on-channels) and paste it on this property to "link" the cards. Add multiple URL properties to link more items to the card.

The 2nd workaround could make the job only if :
1.  Instead of displaying the URL, it would display the card's title
2. It would be possible to group by URL properties (in board & table view)

**Comment by PatrickHuetter** [comment]
> Hi everyone, Good to know that this feature starts to exist. Coming from Jira software I definitely miss the Master task / sub tasks links. I use this a lot because I split my project into user stories which are features with user oriented description and acceptance criteria. Then those user stories are split into technical tasks with technical description and acceptance criteria ; which are not all assigne to the same person
> 
> Those come as one card that can be extended to see the full progress of the user story. Here is a capture of the agile board I use. <img alt="Screenshot 2022-07-20 at 16 29 38" width="1489" src="https://user-images.githubusercontent.com/1583340/180008072-347875ef-d9be-48e0-9d48-93e10aed514c.png"> Notice on this second picture that cards that have no subtasks are simple cards (below) <img alt="Screenshot 2022-07-20 at 16 31 11" width="1489" src="https://user-images.githubusercontent.com/1583340/180008511-788aff7d-9712-44dd-8bac-02aed9815057.png">
> 
> One point about this approach is that I use two boards to manage our project. The kanban board depicted on the pictures is associated to a sprint and only contains tasks this specific sprint. The backlog board that contains all the taks that will have to be done in the future and are not placed in any sprint.
> 
> Hope that this will bring some ideas to this awesome project developers.

I can confirm that. Linking cards with further meta information about the link would be very nice since we also use this a lot within JIRA to show dependencies and related tasks or blocking tasks.

**Comment by olivierlambert** [comment]
The same for me, this would be enough to us to replace our existing Kanban solution :) (wekan)

Keep up the good work!
